versionedRouter.js

```diff
@@ -123,10 +150,32 @@ async function handleCdkV2(destName, parsedEvent, flowType) {
         ...errObj.statTags
       }
     };
+
+    errNotificationClient.notify(err, `CDK v2`, {
+      ...resp,
+      // TODO: add common metadata here
+      ...getReqMetadata()
+    });
+
+    return resp;
   }
 }
 
 async function handleDest(ctx, version, destination) {
+  const getReqMetadata = event => {
+    try {
```

Comment:
can we do this instead of wrapping in try catch?
```js
return {
         destType: destination,
         destinationId: event?.destination?.ID,
         destName: event?.destination?.Name,
         metadata: event?.metadata
       };
```

Reply:
I wrapped all the logic related to error notification in try-catches even though unnecessary for most of the cases as I didn't want anything going wrong to affect actual processing. For the same reason, the instrumentation is usually the last thing in the catch blocks.